### PR TITLE
Routing refactoring, preparations related to new navigation

### PR DESCRIPTION
**Pull request by rasom**
The main purpose of PR is to introduce a new structure for routing definition which will allow to make changes necessary for a new navigation with touching smaller parts of existing code.

- screens definition is separated from configuration
  (deduplicates screens which are used among several stacks)
- all four stacks moved to separate namespaces
- configuration is just a data

other notes:
- old routing is not removed to make pr a bit smaller, will be done in a separate commit

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/routing/routes.cljs [hunk 1]
```diff
@@ -0,0 +1,151 @@
+(ns status-im.ui.screens.routing.routes
+  (:require
+   [status-im.ui.screens.main-tabs.views :as main-tabs]
+
+   [status-im.ui.screens.accounts.login.views :refer [login]]
```
Comment (yenda) [inline src/status_im/ui/screens/routing/routes.cljs]: The old code was using refer as well but shouldn't we use :as for new code

**Review by yenda — COMMENTED** [review]
Could you add some doctstrings to the functions in `routing/core.cljs`

File: src/status_im/ui/screens/routing/routes.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/ui/screens/routing/routes.cljs]: technically that's a copied/moved old code. But ok

**Comment by rasom** [comment]
@yenda some docstrings are added

**Review by yenda — APPROVED** [review]

**Comment by Serhy** [comment]
@rasom e2e tests are green since new routing is not enabled for e2e build.

But if it's enabled (NEW ROUTING = 1), Status app fails to start. 

screenshot 2019-02-14 at 12 32 56

Logcat snippet:
```
02-14 12:20:03.306 18255 18313 I ReactNativeJS: NAME: nil
02-14 12:20:03.306 18255 18313 I ReactNativeJS: MESSAGE: "Screen :login is not defined."
02-14 12:20:03.306 18255 18313 I ReactNativeJS: 
02-14 12:20:03.306 18255 18313 I ReactNativeJS: 
02-14 12:20:03.306 18255 18313 I ReactNativeJS: 
02-14 12:20:03.306 18255 18313 I ReactNativeJS: ***
02-14 12:20:03.309 18255 18314 E unknown:ReactNative: Screen :login is not defined.
02-14 12:20:03.309 18255 18314 E unknown:ReactNative: 
02-14 12:20:03.309 18255 18314 E unknown:ReactNative: This error is located at:
02-14 12:20:03.309 18255 18314 E unknown:ReactNative:     in reagent2
02-14 12:20:03.309 18255 18314 E unknown:ReactNative:     in root
02-14 12:20:03.309 18255 18314 E unknown:ReactNative:     in reagent1
02-14 12:20:03.309 18255 18314 E unknown:ReactNative:     in RCTView
02-14 12:20:03.309 18255 18314 E unknown:ReactNative:     in RCTView

```


Taking into account new routing behind the flag, anyway - is it expected for this PR?

**Pull request merged**